pulsar-broker/src/main/java/com/yahoo/pulsar/broker/loadbalance/BrokerHostUsage.java

```diff
@@ -71,25 +91,152 @@ public BrokerHostUsage(PulsarService pulsar) {
+        } else {
+            double elapsedSeconds = (now - lastCollection) / 1000d;
+            Double nicUsageTx = (totalNicUsageTx - lastTotalNicUsageTx) / elapsedSeconds;
+            Double nicUsageRx = (totalNicUsageRx - lastTotalNicUsageRx) / elapsedSeconds;
+            Long cpuTimeDiff = cpuStat.getTotalTime() - lastCpuStat.getTotalTime();
+            Long cpuUsageDiff = cpuStat.getUsage() - lastCpuStat.getUsage();
+            Double cpuUsage = ((double) cpuUsageDiff / (double) cpuTimeDiff) * totalCpuLimit;
+            usage.setMemory(new ResourceUsage(memUsage, totalMemLimit));
+            usage.setBandwidthIn(new ResourceUsage(nicUsageRx, totalNicLimit));
+            usage.setBandwidthOut(new ResourceUsage(nicUsageTx, totalNicLimit));
+            usage.setCpu(new ResourceUsage(cpuUsage, totalCpuLimit));
+        }
+
+        lastTotalNicUsageTx = totalNicUsageTx;
+        lastTotalNicUsageRx = totalNicUsageRx;
+        lastCpuStat = cpuStat;
+        lastCollection = System.currentTimeMillis();
+        return usage;
+    }
+
+    private double getTotalCpuLimit() {
+        return (double) (100 * Runtime.getRuntime().availableProcessors());
+    }
+
+    private CpuStat getTotalCpuUsage() {
         try {
-            /**
-             * Spawns a python process and runs the usage exporter script. The script return the machine information in
-             * the json format.
-             */
-
-            int exitCode = CmdUtility.exec(writer, usageScriptPath, "--host-usage-check-interval",
-                    Integer.toString(hostUsageCheckInterval));
-            if (exitCode != 0) {
-                LOG.warn("Process exited with non-zero exit code - [{}], stderr - [{}] ", exitCode, writer.toString());
-                throw new IOException(writer.toString());
+            String[] words = Files.lines(Paths.get("/proc/stat"))
+                    .findFirst()
+                    .get().split("\\s+");
+            LongStream values = Arrays.stream(words)
+                    .filter(s -> !s.contains("cpu"))
+                    .mapToLong(Long::parseLong);
+
+            return new CpuStat(values.sum(), values.limit(3).sum());
+        } catch (IOException e) {
+            return new CpuStat(0L, 0L);
+        }
+    }
+
+    private double getTotalMemLimit() {
+        return ((double) systemBean.getTotalPhysicalMemorySize()) / (1024 * 1024);
+    }
+
+    private double getTotalMemUsage() {
+        Double free = ((double) systemBean.getFreePhysicalMemorySize()) / (1024 * 1024);
+        return totalMemLimit - free;
+    }
+
+    private List<String> getNics() {
+        try {
+            return Files.list(Paths.get("/sys/class/net/"))
+                    .filter(this::isPhysicalNic)
+                    .map(path -> path.getFileName().toString())
+                    .collect(Collectors.toList());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Same here, we should either propagate the IOException or at least print a warning log message if we can't read the information.